intl_translation_format/lib/test/testers.dart

```diff
@@ -0,0 +1,348 @@
+import 'package:intl_translation_format/intl_translation_format.dart';
+import 'package:intl_translation/src/intl_message.dart';
+import 'package:test/test.dart';
+import 'format_test.dart';
+
+void testFormat(FormatTester tester) {
+  tester._test();
+}
+
+/// An abstract class that allows to test a [TranslationFormat] into a standarized
+/// test with diverse messages.
+///
+/// It is recommended to use the implmentations [MonolingualFormatTester] for
+/// monolingual formats, and [MultilingualParsingTester],
+/// [MultilingualGenerateTemplateTester] for formats with multiple locales.
+///
+/// See [intl_translation_arb](https://github.com/jamesblasco/intl_translation_format/blob/master/intl_translation_arb/test/intl_translation_arb_test.dart)
+/// for an actual implementation of a FormatTester.
+///
+abstract class FormatTester<T> {
+  TranslationFormat get format;
+  String get defaultLocale => 'en';
+
+  Map<String, MainMessage> get messages {
+    return {
+      'simpleMessage': IcuMainMessage('Simple Message', 'simpleMessage'),
+      'messageWithMetadata':
+          IcuMainMessage('Message With Metadata', 'messageWithMetadata')
+            ..description = 'This is a description',
+      'pluralExample': IcuMainMessage(
+          '{howMany,plural, =0{No items}=1{One item}many{A lot of items}other{{howMany} items}}',
+          'pluralExample'),
+      'messageWithVariable':
+          IcuMainMessage('Share {variable}', 'messageWithVariable'),
+    };
+  }
+
+  Map<String, MainMessage> get translatedMessages {
+    return {
+      'simpleMessage': IcuMainMessage('Mensaje simple', 'simpleMessage'),
+      'messageWithMetadata':
+          IcuMainMessage('Mensaje con Metadatos', 'messageWithMetadata')
+            ..description = 'This is a description',
+      'pluralExample': IcuMainMessage(
+          '{howMany,plural, =0{Ningún elemento}=1{Un elemento}many{Muchos elementos}other{{howMany} elementos}}',
+          'pluralExample'),
+      'messageWithVariable':
+          IcuMainMessage('Compartir {variable}', 'messageWithVariable'),
+    };
+  }
+
+  _test();
+}
+
+abstract class ParsingFormatTester<T> implements FormatTester<T> {
```

Comment:
These could use comments. I'm not clear why these implement and then concrete tests also implement these? Couldn't one or the other extend?